receiver/podmanreceiver/libpod_data.go

```diff
@@ -0,0 +1,85 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
+// +build !windows
+
+package podmanreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/podmanreceiver"
+
+import "time"
+
+type Container struct {
```

Comment:
Why all these types need to be exposed now?

Reply:
The main idea is to move all the logic to `internal/podman` like `internal/docker` in a future PR. Then the receiver would just call the exported functions of this package that in some cases return these types (e.g `Containers() []Container`). I think that the linter will complain if an exposed function does not return an exposed type. 

Nonetheless, if you think that it would be better to keep these types and the corresponding functions private, we can move their exposure in another PR.

Reply:
Why do you want to move it to `internal/podman`? Is there any other component that will depend on it?

Reply:
Just to have the logic of a generic container scraper (for podman and docker) in a hypothetical `internal/container`. But I agree with you, for this PR these type should remain unexposed.
Switch to unexported fields in: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10421/commits/39b37844f9a5f7fecf9206844e0ff3bc03953856